src/WebApi_ws_console.cpp

```diff
@@ -21,16 +21,30 @@ void WebApiWsConsoleClass::init(AsyncWebServer& server, Scheduler& scheduler)
 
     scheduler.addTask(_wsCleanupTask);
     _wsCleanupTask.enable();
+
+    _simpleDigestAuth.setUsername(AUTH_USERNAME);
+    _simpleDigestAuth.setRealm("console websocket");
+
+    reload();
+}
+
+void WebApiWsConsoleClass::reload()
+{
+    _ws.removeMiddleware(&_simpleDigestAuth);
```

Comment:
you don't need to remove the middleware: you can just call `_simpleDigestAuth.setAuthMethod(NO_AUTH)`
(I will release a new version soon once the PR is merged: https://github.com/mathieucarbou/ESPAsyncWebServer/pull/113)

Reply:
Thanks for commenting here, @mathieucarbou. Your suggestion makes a lot of sense. Could you answer us this: Do we even need multiple instances of the AuthenticationMiddleware if we want to protect all websockets the same? Can we have a single instance and add it to all servers? I guess there will be no problem with BASIC auth, but DIGEST possibly needs request-specific state, so I am not sure whether a static instance can just handle all connections?

Reply:
One single instance added to the server only will protected globally all handlers (http, ws, sse). But only the initial connection creation!